src/utils/simplify-youtube-response.ts

```diff
@@ -0,0 +1,13 @@
+import { YoutubeResponse } from '../interfaces/responses/youtube-response';
+import { YoutubeVideo } from '../classes/youtube-video';
+
+export const simplifyYoutubeResponse = (
+    responseData: YoutubeResponse
+): YoutubeVideo => {
+    const video = responseData.snippet;
+    // Video publish date is stored in "contentDetails", publish date in "snippet"
+    // is for when this video was added to the playlist
+
+    const publishedDate = responseData.contentDetails?.videoPublishedAt;
```

Comment:
Let's move this logic into the `class` definition for YouTube. This is part of youtube-specific logic we can reuse every time we want to construct the class and so it should belong there.